app/process/handler/consensus-message/consensus-message.go

```diff
@@ -89,21 +111,95 @@ func (cmh ConsensusMessageHandler) handlePayload(payload []byte) error {
 	}
 
 	err = cmh.repository.Create(&message.TransactionMessage{
-		TransactionId: m.TransactionId,
-		EthAddress:    m.EthAddress,
-		Amount:        m.Amount,
-		Fee:           m.Fee,
-		Signature:     m.Signature,
-		Hash:          hexHash,
+		TransactionId:        m.TransactionId,
+		EthAddress:           m.EthAddress,
+		Amount:               m.Amount,
+		Fee:                  m.Fee,
+		Signature:            m.Signature,
+		Hash:                 hexHash,
+		SignerAddress:        address.String(),
+		TransactionTimestamp: m.TransactionTimestamp,
 	})
 	if err != nil {
 		return errors.New(fmt.Sprintf("Could not add Transaction Message with Transaction Id and Signature - [%s]-[%s]", m.TransactionId, m.Signature))
 	}
 
-	log.Printf("Successfully verified and persisted TX with ID [%s]\n", m.TransactionId)
+	log.Infof("Successfully verified and persisted TX with ID [%s]", m.TransactionId)
+
+	txSignatures, err := cmh.repository.GetByTransactionId(m.TransactionId, hexHash)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Could not retrieve Transaction Signatures for Transaction [%s]", m.TransactionId))
+	}
+
+	requiredSigCount := len(cmh.validAddresses)/2 + len(cmh.validAddresses)%2
+	log.Infof("Required signatures: [%v]", requiredSigCount)
+
+	txSignatures, err = cmh.repository.GetByTransactionId(m.TransactionId, hexHash)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Could not retrieve Transaction Signatures for Transaction [%s]", m.TransactionId))
+	}
+
+	if len(txSignatures) < requiredSigCount {
+		log.Warnf(fmt.Sprintf("Insignificant amount of Transaction Signatures for Transaction [%s] - [%d] signatures", m.TransactionId, len(txSignatures)))
+		return nil
+	}
+
+	pos := cmh.findMyPosition(txSignatures)
+	if pos == -1 {
+		return errors.New(fmt.Sprintf("Operator is not amongst the potential leaders - [%v]", pos))
+	}
+
+	log.Infof("My position: [%v]", pos)
+
+	now := time.Now()
+	deadline := now.Add(time.Minute * time.Duration(cmh.deadline*pos))
+
+	log.Infof("NOW: [%v]", now)
+	log.Infof("DEADLINE: [%v]", deadline)
+
+	for now.Before(deadline) {
+		if cmh.transactionSent() {
+			return nil
+		}
+		now = time.Now()
+		time.Sleep(10 * time.Second)
+	}
+	log.Infof("I should submit the transaction now!")
+
+	tasm := &validatorproto.TopicAggregatedSignaturesMessage{
+		Hash:      hexHash,
+		EthTxHash: "0x12345",
+	}
+
+	_, err = proto.Marshal(tasm) // topicMessageBytes
+	if err != nil {
+		log.Error("Could not marshal Topic Aggregated Signatures Message [%s] - [%s]", tasm, err)
+	}
+
+	log.Infof("I should send the message! - [%v]", tasm)
```

Comment:
What if this operator was the 2nd leader and the 1st one submitted it?
Once someone submits the TransactionHash to the topicID, this flow should be able to figure out that the transaction was already submitted and should stop the leader election for this tx (`leaderElection.stop(txId)`)